api/src/infrastructure/sql-adapter.ts

```diff
@@ -47,27 +45,36 @@ export class SQLAdapter {
   public generateSqlFromWidgetDataFilters(filters: WidgetDataFilters): string {
     let filterClause: string = 'WHERE ';
     for (const filter of filters) {
-      const columnValues = PageFilterQuestionMap.getColumnValueByFilterName(
-        filter.name,
-      );
-      if (columnValues === undefined) {
-        this.logger.warn(
-          `${filter.name} not found in PageFilterQuestionMap. Skipping filter...`,
-          this.constructor.name,
-        );
-        continue;
-      }
-      for (const value of columnValues) {
-        filterClause += `(hierarchy_level_2 = '${value}' AND (`;
-
+      // Countries
+      if (filter.name == 'eu-member-state') {
+        filterClause += '(';
         for (const filterValue of filter.values) {
-          filterClause += `categorical_answer ${filter.operator} '${filterValue}' OR `;
+          filterClause += `country_code ${filter.operator} '${CountryISO3Map.getISO3ByCountryName(filterValue)}' OR `;
         }
         filterClause = filterClause.slice(0, -4);
-        filterClause += ')) AND ';
+        filterClause += ') AND ';
+        continue;
+      }
+
+      filterClause += `(question_indicator = '${filter.name}' AND (`;
+
+      for (const filterValue of filter.values) {
+        filterClause += `answer ${filter.operator} '${filterValue}' OR `;
       }
+      filterClause = filterClause.slice(0, -4);
+      filterClause += ')) AND ';
     }
     filterClause = filterClause.slice(0, -4);
     return filterClause;
   }
+
+  public appendExpressionToFilterClause(
+    filterClause: string,
```

Comment:
I am not sure if this was needed for the current implementation of the model, but we will be removing it in the next task as well